### ISSUE TITLE
Connection::trimObjectName and craft 2 => 3 migration (locales => sites)

### ISSUE THREAD

**Issue by mharj** [craftcms/cms#7153]
### Description
trimObjectName breaks FKs on craft migration (m160807_144858_sites) on mysql/mariadb when table names are long ones (like sproutformscontent ones).
I think trimObjectName should not touch actual table name as FK are unique globally (on mysql/mariadb) and breaks easily in case like 
```
craft_sproutformscontent_somenicelongformsetupen 
craft_sproutformscontent_somenicelongformsetupdk 
craft_sproutformscontent_somenicelongformsetupit
```

Also noticed that it's bit strange that getTableNameWithoutPrefix not dropping out actually prefix which leads prefix_prefix_table_name_..... FK names.

### Steps to reproduce

1. Import mysql/mariadb craft 2 database with long table names with FK to locales
2. Run migration to latest craft

### Additional info

- Craft version: 3.5.12.1
- PHP version: 7.4.3
- Database driver & version: mysql 8.0.22
- Plugins & versions: not needed

### possible fixes
- trimObjectName to use something like plain "uuid-xxxx-xx-xxxxx-xxx-xx-fk" in case it's too long
- m160807_144858_sites to use $refTable.'_siteId_fk' as FK (our current temp hack)
- someway to double check that FK is not existing and do some magic to create bit different one

If someone having same FK issue with long table names, current temp hack goes to craft/vendor/craftcms/cms/src/migrations/m160807_144858_sites.php
around line: 287
```php
                // Add a new *__siteId column + FK for each column in this FK that points to locales.locale
                foreach ($columns as $refColumn) {
                    $newColumn = $refColumn . '__siteId';
                    $isNotNull = !$originalRefTable->getColumn($refColumn)->allowNull;
                    $this->addSiteColumn($refTable, $newColumn, $isNotNull, $refColumn);
                    $this->addForeignKey($refTable.'_LS_fk', $refTable, [$newColumn], Table::SITES, ['id'], 'CASCADE', 'CASCADE');
                }
```

**Comment by brandonkelly** [comment]
Just fixed this for Craft 3.6. Going forward all index/PK/FK names will be completely random, rather than based on a given table name, etc.

**Comment by brandonkelly** [comment]
Craft 3.6 has now been officially released ✨
